lock-api/src/main/java/com/palantir/lock/watch/ClientLockWatchSnapshotUpdaterImpl.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.watch;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.lock.LockDescriptor;
+import com.palantir.logsafe.Preconditions;
+
+final class ClientLockWatchSnapshotUpdaterImpl implements ClientLockWatchSnapshotUpdater {
+    private final Set<LockWatchReferences.LockWatchReference> watches;
+    private final Set<LockDescriptor> locked;
+    private final EventVisitor visitor;
+    private Optional<IdentifiedVersion> snapshotVersion;
```

Comment:
Turns out it makes sense for the snapshot to know the version that it has processed thus far. This version is essentially the latest version that is no longer in the log.